posthog/hogql/database/database.py

```diff
@@ -491,6 +492,14 @@ def create_hogql_database(
     self_managed_warehouse_tables: TableStore = {}
     views: TableStore = {}
 
+    if skip_dw_tables:
+        # Skip data warehouse table loading and proceed directly to adding empty tables
+        # This is used internally for QueryRunners that know ahead of time that they will not be using the data warehouse tables
+        database.add_warehouse_tables(**warehouse_tables)
+        database.add_warehouse_self_managed_tables(**self_managed_warehouse_tables)
+        database.add_views(**views)
```

Comment:
So, it's not creating the tables that make it slow (you're not skipping that, just skipping a simple append to an array), but rather the fact that we need to iterate over them when resolving our tables, is that it?

If that's the case, there's also some room for improvement on our HogQL code by resolving references in a smarter way (look at built-in tables before looking at data warehouse)

Reply:
I suggest you also spend some time skipping the creation of these tables/view up above, that should translate to an even bigger improvement.